pkg/controller/jobframework/reconciler.go

```diff
@@ -136,11 +137,19 @@ func NewReconciler(
 	}
 }
 
-func (r *JobReconciler) ReconcileGenericJob(ctx context.Context, req ctrl.Request, job GenericJob) (ctrl.Result, error) {
-	object := job.Object()
+func (r *JobReconciler) ReconcileGenericJobWrapper(ctx context.Context, req ctrl.Request, job GenericJob) (ctrl.Result, error) {
 	log := ctrl.LoggerFrom(ctx).WithValues("job", req.String(), "gvk", job.GVK())
 	ctx = ctrl.LoggerInto(ctx, log)
 
+	result, err := r.ReconcileGenericJob(ctx, req, job)
+
+	return result, r.ignoreUnretryableError(ctx, err)
```

Comment:
this avoids a look up.

```suggestion
	return result, r.ignoreUnretryableError(log, err)
```